src/client/editor/Editor.js

```diff
@@ -378,6 +378,26 @@ export default class Editor {
     return exportScene(this.scene);
   }
 
+  async extendScene(inheritedURI) {
+    const extendSceneDef = {
+      entities: {},
+      inherits: inheritedURI
+    };
+
+    this.deselect();
+    this._resetHelpers();
```

Comment:
You need `_deleteSceneDependencies()` and `_clearCaches()` in here as well. You're basically creating a new scene, so you can follow `loadNewScene` as an example -- I would try to maintain the same order of operations.